features/include/pcl/features/impl/spin_image.hpp

```diff
@@ -326,14 +327,9 @@ pcl::SpinImageEstimation<PointInT, PointNT, PointOutT>::computeFeature (PointClo
   {
     Eigen::ArrayXXd res = computeSiForPoint (indices_->at (i_input));
 
-    // Copy into the resultant cloud
-    for (Eigen::Index iRow = 0; iRow < res.rows () ; iRow++)
-    {
-      for (Eigen::Index iCol = 0; iCol < res.cols () ; iCol++)
-      {
-        output.points[i_input].histogram[ iRow*res.cols () + iCol ] = static_cast<float> (res (iRow, iCol));
-      }
-    }   
+    // Copy into the resultant cloud, can't use std::copy due to type difference
+    std::transform (res.data (), res.data () + res.size (), output.points[i_input].histogram,
```

Comment:
For me, this is the suspect. Are we certain histogram and `res` have same row/column-majorness? Maybe rever this change and if the problem persists.

Reply:
Eigen is [column major](https://eigen.tuxfamily.org/dox/group__TopicStorageOrders.html) by default. But histogram is row major. That's a nice catch.